test/functional/test-3p.js

```diff
@@ -167,6 +168,48 @@ describe('3p', () => {
         /xxxxxx must contain exactly one of attributes: red, green, blue./);
   });
 
-
+  it('should do work only once', () => {
```

Comment:
What I mean is that if `master` calls `computeInMasterFrame` with the same `taskId`, it will call `work` again (not to mention pushing `cb` into the array, which will immediately call it with the old `results` and try to for-loop an object).